HandsOnProject/app/src/main/java/com/example/handsonproject/MainActivity.kt

```diff
@@ -1,11 +1,144 @@
 package com.example.handsonproject
 
-import androidx.appcompat.app.AppCompatActivity
+
 import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.example.handsonproject.databinding.ActivityMainBinding
+import com.example.handsonproject.databinding.ActivityMainBinding.inflate
+
 
 class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+    private var op : Int = 0
+
     override fun onCreate(savedInstanceState: Bundle?) {
+
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        binding = inflate(LayoutInflater.from(this))
+        setContentView(binding.root)
+        with(binding){
+            editText1.hint="angka"
+            textView.text=""
+        }
+    }
+
+
+    fun onTapB1(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "1") }
+    }
+    fun onTapB2(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "2") }
+    }
+    fun onTapB3(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "3") }
+    }
+    fun onTapB4(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "4") }
+    }
+    fun onTapB5(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "5") }
+    }
+    fun onTapB6(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "6") }
+    }
+    fun onTapB7(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "7") }
+    }
+    fun onTapB8(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "8") }
+    }
+    fun onTapB9(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "9") }
+    }
+    fun onTapB0(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "0") }
+    }
+    fun onTapBEqual(view: View){
+        with(binding){
+            if (editText1.text.toString().equals("")&&textView.text.toString().equals("")) return
+            if(op==1){
+                val res=editText1.text.toString().toDouble()+textView.text.toString().toDouble()
+                editText1.setText(res.toString())
+                textView.setText("")
```

Comment:
I got some error when doing this, it says that it expect the input to be Editable, but what is there is String
![Screenshot from 2021-07-28 09-51-39](https://user-images.githubusercontent.com/61104176/127256028-094a59eb-0025-4a3b-a300-e8700ff57d5c.png)

Reply:
@shovian Since Yamauchi-san is on vacation today, I will reply on his behalf.

I have confirmed that the suggested change will work for `stash` but not for `input` :

<img width="280" alt="Screen Shot 2021-07-28 at 12 03 27" src="https://user-images.githubusercontent.com/3959480/127257032-c9fe486b-e41d-4da5-a55f-8745a7d29c22.png">

The reason for that is that `stash` and `input` are of different types: `stash` is a `TextView` whereas `input` is a `TextInputEditText`:

```
        <TextView
            android:id="@+id/stash"
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            android:text="@string/textview" />
```

```
            <com.google.android.material.textfield.TextInputEditText
                android:id="@+id/input"
                android:layout_width="match_parent"
                android:layout_height="wrap_content" />
```

And apparently only the `TextView` does have a `text` property. It seems for `TextInputEditText` you will still have to use the setter.

@atsushi-ageet Once you are back please confirm my analysis.

Reply:
I see, thank you!

Reply:
@atsushi-ageet もうマージ済みですがコメントをいただければと思います。

Reply:
EditTextViewはEditableを返すgetterを持っているため、kotlinはEditableのpropertyとして扱います。

https://stackoverflow.com/a/37374301

そのため、EditTextViewの場合にはsetTextを利用する必要があります。

Reply:
ありがとうございました。